tilavarauspalvelu/models/user/actions.py

```diff
@@ -187,3 +188,27 @@ def can_anonymize(self) -> UserAnonymizationInfo:
             has_open_applications=has_open_applications,
             has_open_payments=has_open_payments,
         )
+
+    @property
+    def is_ad_user(self) -> bool:
+        id_token = self.user.id_token
+        return id_token is not None and id_token.is_ad_login
+
+    @property
+    def is_profile_user(self) -> bool:
+        id_token = self.user.id_token
+        return id_token is not None and id_token.is_profile_login
+
+    @property
+    def user_age(self) -> int | None:
+        birthday = self.user.date_of_birth
+        if birthday is None:
+            return None
+        return relativedelta(local_date(), birthday).years
+
+    @property
+    def is_of_age(self) -> bool:
+        user_age = self.user_age
+        if user_age is None:
+            return False
+        return user_age >= settings.USER_IS_ADULT_AT_AGE
```

Comment:
My intuition is saying that these would fit better directly to the `User` model, since they are properties and don't use data from any other source than itself 🤔 
(No need to make any changes if you prefer them here)

Reply:
Maybe, but I didn't want to bloat the user model any more than it already is.